openlibrary/plugins/openlibrary/home.py

```diff
@@ -51,22 +51,32 @@ def get_homepage():
     )
     return dict(page)
 
-def get_cached_homepage():
+
+def get_cached_homepage(lang='en'):
     five_minutes = 5 * dateutil.MINUTE_SECS
-    lang = web.ctx.get("lang", "en")
     return cache.memcache_memoize(
-        get_homepage, "home.homepage." + lang, timeout=five_minutes)()
+        get_homepage, 'home.homepage.' + lang, timeout=five_minutes
+    )()
+
 
 class home(delegate.page):
     path = "/"
 
     def GET(self):
-        cached_homepage = get_cached_homepage()
+        # Get the locale from the cookie or -if not found- set it to the default ('en')
+        locale_cookie = web.cookies().get('i18n_code')
+        if locale_cookie is not None:
+            locale = locale_cookie
+        else:
+            web.setcookie('i18n_code', 'en', secure=False)
```

Comment:
If the user hasn't selected a language yet, the default should be the highest priority language from their browser's `Accept-Language` header which is supported by OpenLibrary. https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language

There's no reason to force them to start with English when they've already expressed their language preferences.

Reply:
Ok, I'll look into it and provide a fix for this. 

If the highest priority language is not part of the supported set of languages (EN, DE, CS, ES, FR, TE) the setting should default to English, right?

Reply:
Hi @tfmorris. I've implemented an updated fix where the language is selected from the corresponding header. If no OL supporting language is found or if no `Accept-Language` header is sent then, English is used. Is this OK? Or is something more that you want me to do before submitting the updated PR?

Reply:
English is fine as a fallback if OpenLibrary doesn't support any of their preferred languages.